homeassistant/components/alarm_control_panel/xiaomi_gateway.py

```diff
@@ -0,0 +1,230 @@
+        self._turn_on_command = config.get(CONF_TURN_ON_COMMAND)
+        self._turn_on_parameters = config.get(CONF_TURN_ON_PARAMETERS)
+        self._turn_off_command = config.get(CONF_TURN_OFF_COMMAND)
+        self._turn_off_parameters = config.get(CONF_TURN_OFF_PARAMETERS)
+        self._state_property = config.get(CONF_STATE_PROPERTY)
+        self._state_on_value = config.get(CONF_STATE_ON_VALUE).pop()
+        self._state_off_value = config.get(CONF_STATE_OFF_VALUE).pop()
+        self._update_instant = config.get(CONF_UPDATE_INSTANT)
+        self._skip_update = False
+        self._model = device_info.model
+        self._unique_id = "{}-{}-{}".format(device_info.model,
+                                            device_info.mac_address,
+                                            self._state_property)
+        self._icon = 'mdi:flask-outline'
+        self._available = None
+        self._state = None
+        self._state_attrs = {
+            ATTR_MODEL: self._model,
+            ATTR_FIRMWARE_VERSION: device_info.firmware_version,
+            ATTR_HARDWARE_VERSION: device_info.hardware_version,
+            ATTR_STATE_PROPERTY: self._state_property
+        }
+
+    @property
+    def should_poll(self):
+        """Poll the miio device."""
+        return True
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return self._available
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state_attrs
+
+    def update(self):
+        """Return the state of the device."""
+        return self._state
+
+    async def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a device command handling error messages."""
+        from miio import DeviceException
+        try:
+            result = await self.hass.async_add_job(
+                partial(func, *args, **kwargs))
+
+            _LOGGER.info("Response received from miio device: %s", result)
+
+            return result == SUCCESS
+        except DeviceException as exc:
+            _LOGGER.error(mask_error, exc)
+            return False
+
+    async def async_alarm_arm_away(self, code=None):
+        """Turn on."""
+        result = await self._try_command(
+            "Turning the miio device on failed.", self._device.send,
+            'set_arming', ['on'])
```

Comment:
This needs to be implemented in the backend lib, I just pushed my old code online if you are interested in working on it: https://github.com/rytilahti/python-miio/pull/470